### PR TITLE
Fix Style injection caused tailwind override css priorities

### LINKED ISSUE #128: Style injection caused tailwind override css priorities
Novel will inject CSS directly into Root, included tailwindcss. Which could potentially override stylesheets priorities when you have tailwind in the project yourself.

### PR DESCRIPTION
**Pull request by eyalcohen4**
This PR should fix #128 by prefixing the tailwind classes of Novel.

One current Issue I face that blocking a merge is that the `Prose` classes inside the `tiptap/react` editor are not customizable, and therefor the prose class doesn't apply to them.

Screenshot 2023-09-13 at 19 16 36

### CODE REVIEW

File: packages/core/src/styles/prosemirror.css [hunk 1]
```diff
@@ -150,7 +150,7 @@ ul[data-type="taskList"] li[data-checked="true"] > div > p {
   }
 
   &:active {
-    background-color: var(--novel-stone-200);
+    background-color: var(--nnovel-stone-200);
```
Comment (MarkLyck) [inline packages/core/src/styles/prosemirror.css:153]: Is this supposed to have two `n`'s?

Comment (eyalcohen4) [reply packages/core/src/styles/prosemirror.css:153]: nope great catch!

**Comment by alvin** [comment]
@eyalcohen4 not sure if this is the same editor text styling issue you were getting, but after compiling this branch to fix the overall style pollution (which resulted in clean outer site content -- but unstyled editor content) I had success passing in editorProps (which Novel exposes) as documented [on the tiptap site](https://tiptap.dev/guide/styling#with-tailwind-css): 

```
      <NovelEditor
        editorProps={{
          attributes: {
            class: 'prose dark:prose-invert prose-sm sm:prose-base lg:prose-lg focus:outline-none ',
          },
        }}
        ...
        />
  ```

So, it appears that using branch plus that trick, plus adding `.ProseMirror` scoped h1, h2, h3 styles in `prosemirror.css` has this one unblocked for me.    Thanks for the fix!

**Comment by steven-tey** [comment]
Thanks so much for this PR @eyalcohen4!! Also fixed the prose issue (without needing to use the exposed `editorProps` prop from `<Novel/>`)

CleanShot 2023-10-03 at 09 18 54

To be honest this doesn't feel like a very "clean" fix to me 😅 – I've been cracking my head trying to find a better solution, but to no avail so far.

However, it does the job perfectly (thank you @alvin for confirming) and it's also not too hard to revert in the future if we do find a better fix!

Will cut a new release soon after merging a few other PRs!

**Pull request merged**